java/solrmarc_mixin/src/de/unituebingen/ub/ubtools/solrmarcMixin/TuelibMixin.java

```diff
@@ -3285,4 +3285,68 @@ public String extractFirstK10PlusPPNAndTitle(final Record record, final String f
 
         return null;
     }
+
+    public static Long getRangeMaximum(final Record record, final String rangeFieldTag) {
+        final DataField rangeField = (DataField) record.getVariableField(rangeFieldTag);
+        if (rangeField == null)
+            return null;
+
+        final Subfield subfieldA = rangeField.getSubfield('a');
+        if (subfieldA == null)
+            return null;
+
+        final String[] parts = subfieldA.getData().split(",");
+
+        long maximum = Long.MIN_VALUE;
+        for (final String part : parts) {
+            final String[] range = part.split(":");
+            if (range.length != 2) {
+                System.err.println(range + " is not a valid range! (1)");
+                System.exit(-1);
+            }
+
+            try {
+                final long upper = Long.parseLong(range[1]);
+                if (upper > maximum)
+                    maximum = upper;
+            } catch (NumberFormatException e) {
+                System.err.println(range + " is not a valid range! (2)");
+                System.exit(-1);
+            }
+        }
+
+        return maximum;
+    }
+
+    public static Long getRangeMinimum(final Record record, final String rangeFieldTag) {
+        final DataField rangeField = (DataField) record.getVariableField(rangeFieldTag);
+        if (rangeField == null)
+            return null;
+
+        final Subfield subfieldA = rangeField.getSubfield('a');
+        if (subfieldA == null)
+            return null;
+
+        final String[] parts = subfieldA.getData().split(",");
+
+        long minimum = Long.MAX_VALUE;
+        for (final String part : parts) {
+            final String[] range = part.split(":");
+            if (range.length != 2) {
+                System.err.println(range + " is not a valid range! (3)");
+                System.exit(-1);
+            }
+
+            try {
+                final long lower = Long.parseLong(range[0]);
+                if (lower > minimum)
```

Comment:
> Long.MAX_VALUE ?

Reply:
Was a cop-'n-paste problem and should be resolved by now.